### PR TITLE
58 incident query

### PR DESCRIPTION
**Pull request by kidbombay**
WIP on some tests that are commented out.

### CODE REVIEW

**Comment by kidbombay** [comment]
This is what I mean by an incident query model.
We can update the search api to return this model instead and page through that.

Benefits
We can return results matching the Dashboard UI for population, negative, positive count
We can catch queries in the future (50 people are asking for same exact query, no need to update the query in real time)
We can store all searches for tracking purposes

@BrandonAnderson 


This is not finished but I wanted to get agreement before moving forward

**Comment by seansu4you87** [comment]
`We can return results matching the Dashboard UI for population, negative, positive count`
Can't we do this regardless of our modeling, by just having a different JSON schema?

`We can catch queries in the future (50 people are asking for same exact query, no need to update the query in real time)`
To provide a caching solution, don't the incidents themselves need to be part of the schema for the `IncidentQuery`?

`We can store all searches for tracking purposes`
For accurate tracking, do we want some column to increment count then, if someone makes a search that has already been made?

**Comment by kidbombay** [comment]
The JSON Schema is similar to the JSON API which we might want to migrate to. Industry standard API stuff doesn't have non meta content in meta. It's only paging info. 

We could explore storing that as well. I'm more concerned with counts and place data. 

Yes adding a count of calls would be good. What should we call that.

File: app/models/incident.rb [hunk 1]
```diff
@@ -21,6 +21,8 @@ class Incident < ApplicationRecord
   scope :by_incident_type, -> (incident_type) { where(incident_type: incident_type) }
   scope :between, ->          (start_date, end_date) { where("incidents.start_time BETWEEN ? and ?", start_date, end_date) }
   scope :with_any_rating, ->  (ratings) { where(rating: ratings.map(&:to_i)) }
+  scope :negative, ->         { where("rating < ?", 4) }
```
Comment (kidbombay) [inline app/models/incident.rb]: Negative is 2 or less
Positive is 4 or higher
Neutral is 3

**Pull request closed without merging**